program/object-detection-tf-py/calc_metrics_oid.py

```diff
@@ -0,0 +1,152 @@
+      # Detections dict can't be empty even there is not detection for this image
+      if cnt == 0:
+        detection['detection_boxes'] = np.array([[0.0, 0.0, 0.0, 0.0]])
+        detection['detection_classes'] = np.array([0])
+        detection['detection_scores'] = np.array([0.0])
+  except Exception:
+    print('Exception when reading data from {}'.format(detection['key'] + '.txt'))
+    detection = None
+  finally:
+    return detection
+
+def get_annotations(annotations, class_map):
+  '''
+  Get annotations from file to dictionary in memory
+
+  Args:
+    annotations: groundtruth (file with annotations)
+    class_map: dictionary for classes, text id => number id 
+
+  Returns:
+    Dictionary with annotations
+  '''
+  res = {}
+
+  with open(annotations, 'r') as ann:
+    ann.readline() #header
+    for line in ann:
+      str = line.split(',')
+      key = str[0]
+      img_class = class_map[str[2]]
+      x_min = float(str[4])
+      x_max = float(str[5])
+      y_min = float(str[6])
+      y_max = float(str[7])
+      group = int(str[10])
+      if key not in res:
+        res[key] = []
+      res[key].append({
+        'class': img_class,
+        'x_min': x_min,
+        'x_max': x_max,
+        'y_min': y_min,
+        'y_max': y_max,
+        'group': group
+      })
+
+  return res
+
+def fill_annotations(detection, annotations):
+  for ann in annotations:
+    detection['groundtruth_group_of'] = np.append(detection['groundtruth_group_of'], ann['group'])
+    detection['groundtruth_boxes'] = np.append(detection['groundtruth_boxes'],
+      [[ann['x_min'], ann['y_min'], ann['x_max'], ann['y_max']]], 0)
+    detection['groundtruth_classes'] = np.append(detection['groundtruth_classes'], ann['class'])
+  return detection
+
+
+def evaluate(res_dir, annotations, label_map_path):
+  '''
+  Calculate OID metrics via evaluator class included in TF models repository
+  https://github.com/tensorflow/models/tree/master/research/object_detection/metrics
+
+  Reads pre-computed object detections and groundtruth.
+
+  Args:
+    res_dir: pre-computed object detections directory
+    annotations: groundtruth (file with annotations)
+    label_map_path: labelmap file
+
+  Returns:
+    Evaluated detections metrics.
+  '''
+  class EvaluatorConfig:
+    metrics_set = ['open_images_V2_detection_metrics']
+
+  eval_config = EvaluatorConfig()
+
+  categories = label_map_util.create_categories_from_labelmap(label_map_path)
+  class_map = label_map_util.get_label_map_dict(label_map_path, False, False)
+
+  object_detection_evaluators = evaluator.get_evaluators(
+      eval_config, categories)
+  # Support a single evaluator
+  object_detection_evaluator = object_detection_evaluators[0]
+
+  print('Loading annotations...')
+  ann = get_annotations(annotations, class_map)
+
+  files = ck_utils.get_files(res_dir)
+  for file_index, file_name in enumerate(files):
+    print('Loading detections: {} of {} ...'.format(file_index+1, len(files)))
```

Comment:
Please, use condition `full_report` to avoid large printouts when a lot of images are detected:
```
if full_report:
      print('Loading detections and annotations for {} ({} of {}) ...'.format(file_name, file_index+1, len(files)))
    elif (file_index+1) % 100 == 0:
      print('Loading detections and annotations: {} of {} ...'.format(file_index+1, len(files)))
```